pkg/volume/util/operationexecutor/operation_generator_test.go

```diff
@@ -0,0 +1,205 @@
+		st_mf := getMetricFamily(metric_family_name)
+		if st_mf == nil {
+			t.Fatalf("Couldn't find the metric family with name: %s", metric_family_name)
+		}
+
+		labels := make(map[string]string)
+		for _, labelPair := range st_mf.GetMetric()[0].Label {
+			labels[*labelPair.Name] = *labelPair.Value
+		}
+		assert.Equal(t, "success", labels["status"])
+		assert.Equal(t, "unmap_volume", labels["operation_name"])
+		assert.Equal(t, expectedPluginName, labels["volume_plugin"])
+		assert.Equal(t, float64(1), *st_mf.GetMetric()[0].Counter.Value)
+	}
+}
+
+func getTestOperationGenerator(volumePluginMgr *volume.VolumePluginMgr) OperationGenerator {
+	fakeKubeClient := controllervolumetesting.CreateTestClient()
+	fakeRecorder := &record.FakeRecorder{}
+	fakeHandler := volumetesting.NewBlockVolumePathHandler()
+	operationGenerator := NewOperationGenerator(
+		fakeKubeClient,
+		volumePluginMgr,
+		fakeRecorder,
+		false, /* checkNodeCapabilitiesBeforeMount */
+		fakeHandler)
+	return operationGenerator
+}
+
+func getTestVolumeToUnmount(podName string, pdName string, pod *v1.Pod) MountedVolume {
+	volumeMode := v1.PersistentVolumeBlock
+	volumeSpec := &volume.Spec{
+		PersistentVolume: &v1.PersistentVolume{
+			Spec: v1.PersistentVolumeSpec{
+				VolumeMode: &volumeMode,
+				PersistentVolumeSource: v1.PersistentVolumeSource{
+					GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+						PDName: "pdvol-1",
+					},
+				}}}}
+	volumeToUnmount := MountedVolume{
+		PodName:    volumetypes.UniquePodName(podName),
+		VolumeName: v1.UniqueVolumeName(pdName),
+		PodUID:     pod.UID,
+		PluginName: gcePdVolumePluginName,
+		VolumeSpec: volumeSpec,
+	}
+	return volumeToUnmount
+}
+
+func getMetricFamily(metric_family_name string) *io_prometheus_client.MetricFamily {
+	metric_families, _ := prometheus.DefaultGatherer.Gather()
+	for _, mf := range metric_families {
+		if *mf.Name == metric_family_name {
+			return mf
+		}
+	}
+	return nil
+}
+
+func initTestPlugins(t *testing.T) (*volume.VolumePluginMgr, volume.VolumePlugin, string) {
+	tmpDir, err := utiltesting.MkTmpdir("csi-test")
+	if err != nil {
+		t.Fatalf("can't create temp dir: %v", err)
+	}
+
+	client := fakeclient.NewSimpleClientset()
+
+	// Start informer for CSIDrivers.
+	factory := informers.NewSharedInformerFactory(client, time.Minute)
+	csiDriverInformer := factory.Storage().V1beta1().CSIDrivers()
+	csiDriverLister := csiDriverInformer.Lister()
+	go factory.Start(wait.NeverStop)
+
+	host := volumetest.NewFakeVolumeHostWithCSINodeName(
+		tmpDir,
+		client,
+		nil,
+		"fakeNode",
+		csiDriverLister,
+	)
+	plugMgr := &volume.VolumePluginMgr{}
+	volumePlugins := gcepd.ProbeVolumePlugins()
+	volumePlugins = append(volumePlugins, csi.ProbeVolumePlugins()...)
+	plugMgr.InitPlugins(volumePlugins, nil /* prober */, host)
+
+	csiplugin, err := plugMgr.FindPluginByName(csi.CSIPluginName)
+	if err != nil {
+		t.Fatalf("can't find plugin %v", csi.CSIPluginName)
```

Comment:
csi.CSIPluginName should be a string? maybe %s is more clear here

Reply:
fixed, thanks!